src/pages/instances/MigrateInstanceModal.tsx

```diff
@@ -0,0 +1,206 @@
+import { fetchClusterMembers } from "api/cluster";
+import Loader from "components/Loader";
+import { pluralize } from "util/instanceBulkActions";
+import { getValidInstancesForMigration } from "util/instanceMigrate";
+import ScrollableTable from "components/ScrollableTable";
+
+interface Props {
+  close: () => void;
+  migrate: (target: string, instances: LxdInstance[]) => void;
+  instances: LxdInstance[];
+}
+
+const MigrateInstanceModal: FC<Props> = ({ close, migrate, instances }) => {
+  const { data: members = [], isLoading } = useQuery({
+    queryKey: [queryKeys.cluster, queryKeys.members],
+    queryFn: fetchClusterMembers,
+  });
+
+  const [selectedMember, setSelectedMember] = useState("");
+
+  const isBulkMigration = instances.length > 1;
+
+  const handleEscKey = (e: KeyboardEvent<HTMLElement>) => {
+    if (e.key === "Escape") {
+      close();
+    }
+  };
+
+  const handleMigrate = (instances: LxdInstance[]) => {
+    migrate(selectedMember, instances);
+    close();
+  };
+
+  const handleCancel = () => {
+    if (selectedMember) {
+      setSelectedMember("");
+      return;
+    }
+
+    close();
+  };
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Roles", sortKey: "roles" },
+    { content: "Architecture", sortKey: "architecture" },
+    { content: "Status", sortKey: "status" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = members.map((member) => {
+    const disableReason =
+      !isBulkMigration && member.server_name === instances[0].location
+        ? "Instance already running on this member"
+        : "";
+
+    const selectMember = () => {
+      if (disableReason) {
+        return;
+      }
+
+      if (selectedMember === member.server_name) {
+        setSelectedMember("");
+        return;
+      }
+
+      setSelectedMember(member.server_name);
+    };
+
+    return {
+      className: "u-row",
+      columns: [
+        {
+          content: (
+            <div
+              className="u-truncate migrate-instance-name"
+              title={member.server_name}
+            >
+              {member.server_name}
+            </div>
+          ),
+          role: "cell",
+          "aria-label": "Name",
+          onClick: selectMember,
+        },
+        {
+          content: member.roles.join(", "),
+          role: "cell",
+          "aria-label": "Roles",
+          onClick: selectMember,
+        },
+        {
+          content: member.architecture,
+          role: "cell",
+          "aria-label": "Architecture",
+          onClick: selectMember,
+        },
+        {
+          content: member.status,
+          role: "cell",
+          "aria-label": "Status",
+          onClick: selectMember,
+        },
+        {
+          content: (
+            <Button
+              onClick={selectMember}
+              dense
+              title={disableReason}
+              disabled={!!disableReason}
```

Comment:
```suggestion
              disabled={Boolean(disableReason)}
```
I think this is easier to read.